controllers/workspace/env/const.go

```diff
@@ -0,0 +1,37 @@
+//
+// Copyright (c) 2019-2021 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+
+package env
+
+const (
+	// DevWorkspaceNamespace contains env var name which value is the current namespace where DevWorkspace CR
+	// and related objects live
+	DevWorkspaceNamespace = "DEVWORKSPACE_NAMESPACE"
+
+	// DevWorkspaceId contains env var name which which value is the .status.devworkspaceId of the related
+	// DevWorkspace CR. It can be used to list related objects with WorkspaceIDLabel selector
+	DevWorkspaceId = "DEVWORKSPACE_ID"
+
+	// DevWorkspaceName contains env var name which value is name of the related DevWorkspace CR.
+	// It can be used to list related objects with WorkspaceNameLabel selector
+	DevWorkspaceName = "DEVWORKSPACE_NAME"
+
+	// DevWorkspaceCreator contains env var name which value is the uid of the identity who created the related devworkspace
+	DevWorkspaceCreator = "DEVWORKSPACE_CREATOR"
+
+	// DevWorkspaceIdleTimeout contains env var name which value is the suggested idle timeout
+	DevWorkspaceIdleTimeout = "DEVWORKSPACE_IDLE_TIMEOUT"
+
+	// DevWorkspaceComponentName contains env var name which indicates from which devfile container component
+	// the container is created from. Note the flattened devfile is used to evaluate it.
+	DevWorkspaceComponentName = "DEVWORKSPACE_COMPONENT_NAME"
```

Comment:
Env var name may be a bit confusing since it's not clear enough if a component is evaluated from DevWorkspace Devfile (like plugin component name) or flattened devfile.
If you have a better proposal or would vote for more specific like `DEVWORKSPACE_CONTAINER_COMPONENT_NAME`, `FLATTENED_DEVWORKSPACE_COMPONENT_NAME`, let us know.

Reply:
I would say it's ok as we can only have ENV variables from containers

Reply:
But process inside the container may be interested in `from which user's component is came`, like k8s component, or plugin.